app/models/admin_menu.rb

```diff
@@ -76,11 +91,28 @@ def self.nested_menu_items(scope_name, nav_item)
       next unless items[:children]&.any?
 
       items[:children].each do |child|
+        # NOTE: [@jeremyf] trying to puzzle this one out.  My read is that if the "grandchild" of the
+        # scope matches the controller, return the parent node (e.g. the item).
         return items if child[:controller] == nav_item
       end
     end
+
+    # Because we're using each loops, with short-circuiting returns, we need to make sure we don't
+    # return the results of `items[:children].each`, which will be `items[:children]`.
+    nil
   end
 
+  # @param request [#path] the request object (which must respond to `#path`)
+  # @return [NilClass] when we don't have a menu representation of the nav_item
+  # @return [Hash] the representation of the nav_item
+  #
+  # @see {.nested_menu_items} for implementation details
+  #
+  # @note This method assumes that the last two slugs of the request's path are the relevant
+  #       information for determining which menu item to return.  In other words, strongly consider
+  #       the impact of having admin routes who's paths are comprised of more than 3 slugs (.e.g. we
```

Comment:
Quick, non-blocking suggestion:
```suggestion
  #       the impact of having admin routes whose paths are comprised of more than 3 slugs (.e.g. we
```